integration-tests/tests/src/tests/dictionary.ts

```diff
@@ -0,0 +1,283 @@
+import { assert, expect } from "chai";
+import Realm from "realm";
+
+import { openRealmBefore } from "../hooks";
+
+type Item<ValueType = Realm.Mixed> = {
+  dict: Realm.Dictionary<ValueType>;
+}
+
+type DictValues = { [key: string]: unknown };
+
+type ValueGenerator = (realm: Realm) => DictValues;
+
+type TypedDictionarySuite = {
+  extraSchema?: Realm.ObjectSchema[];
+  type: string;
+  goodValues: DictValues | ValueGenerator;
+  badValues: DictValues;
+  expectedError: string;
+};
+
+describe("Dictionary", () => {
+  describe("with unconstrained (mixed) values", () => {
+    openRealmBefore({
+      schema: [{
+        name: "Item",
+        properties: { dict: "{}" },
+      }, {
+        name: "Person",
+        properties: { name: "string" },
+      }],
+    });
+    
+    it("can be used as a property type in a schema", function(this: RealmContext) {
+      expect(this.realm.isClosed).equals(false);
+      const dictSchemaProperty = this.realm.schema[0].properties.dict as Realm.ObjectSchemaProperty;
+      expect(typeof dictSchemaProperty).equals("object");
+      expect(dictSchemaProperty.type).equals("dictionary");
+      expect(dictSchemaProperty.objectType).equals("mixed");
+    });
+  
+    it("is instanceof Dictionary", function(this: RealmContext) {
+      this.realm.write(() => {
+        const item = this.realm.create<Item>("Item", {});
+        expect(item.dict instanceof Realm.Dictionary);
```

Comment:
So https://github.com/realm/realm-js/issues/3836 isn't a real issue?

Reply:
I at least cannot reproduce it.